src/Nethermind/Nethermind.Baseline/JsonRpc/BaselineModule.cs

```diff
@@ -0,0 +1,307 @@
+        {
+            for (int i = 0; i < hashes.Length; i++)
+            {
+                if (hashes[i] == Keccak.Zero)
+                {
+                    return Task.FromResult(ResultWrapper<Keccak>.Fail("Cannot insert zero hash", ErrorCodes.InvalidInput));
+                }
+            }
+
+            var txData = _abiEncoder.Encode(AbiEncodingStyle.IncludeSignature, ContractMerkleTree.InsertLeavesAbiSig, new object[] {hashes});
+
+            Transaction tx = new Transaction();
+
+            tx.Value = 0;
+            tx.Data = txData;
+            tx.To = contractAddress;
+            tx.SenderAddress = address;
+            tx.GasLimit = 1000000;
+            tx.GasPrice = 0.GWei();
+
+            Keccak txHash = _txPoolBridge.SendTransaction(tx, TxHandlingOptions.ManagedNonce);
+
+            return Task.FromResult(ResultWrapper<Keccak>.Success(txHash));
+        }
+
+        /// <summary>
+        /// We retrieve the line 3 from here (bytecode) 
+        /// 
+        /// ======= MerkleTreeSHA.sol:MerkleTreeSHA =======
+        /// Binary: 
+        /// 608060405234801561001057600080fd5b5061080980610(...)
+        /// </summary>
+        /// <param name="contract"></param>
+        /// <returns></returns>
+        private async Task<byte[]> GetContractBytecode(string contract)
+        {
+            string[] contractBytecode = await _fileSystem.File.ReadAllLinesAsync($"contracts/{contract}.bin");
+            if (contractBytecode.Length < 4)
+            {
+                throw new IOException("Bytecode not found");
+            }
+
+            if (_logger.IsInfo) _logger.Info($"Loading bytecode of {contractBytecode[1]}");
+            return Bytes.FromHexString(contractBytecode[3]);
+        }
+
+        public async Task<ResultWrapper<Keccak>> baseline_deploy(Address address, string contractType)
+        {
+            byte[] bytecode;
+            try
+            {
+                bytecode = await GetContractBytecode(contractType);
+            }
+            catch (IOException)
+            {
+                return ResultWrapper<Keccak>.Fail($"{contractType} bytecode could not be loaded.", ErrorCodes.ResourceNotFound);
+            }
+
+            Transaction tx = new Transaction();
+            tx.Value = 0;
+            tx.Init = bytecode;
+            tx.GasLimit = 1000000;
+            tx.GasPrice = 20.GWei();
+            tx.SenderAddress = address;
+
+            Keccak txHash = _txPoolBridge.SendTransaction(tx, TxHandlingOptions.ManagedNonce);
+
+            _logger.Info($"Sent transaction at price {tx.GasPrice} to {tx.SenderAddress}");
+            _logger.Info($"Contract {contractType} has been deployed");
+
+            return ResultWrapper<Keccak>.Success(txHash);
+        }
+
+        private BaselineTree RebuildEntireTree(Address treeAddress)
+        {
+            // bad
+
+            Keccak leavesTopic = new Keccak("0x8ec50f97970775682a68d3c6f9caedf60fd82448ea40706b8b65d6c03648b922");
```

Comment:
Please use something like ValidatorContract.CheckInitiateChangeEvent, using AbiDefinition would also help.

Reply:
I would need guidance on that code